tests/bluetooth/bsim_bt/bsim_test_conn_stress/central/src/main.c

```diff
@@ -0,0 +1,929 @@
+				 conn_ref_index);
+			return;
+		}
+
+		while (atomic_test_bit(conn_info_ref->flags, CONN_INFO_LL_DATA_LEN_TX_UPDATED) ==
+		       false) {
+			k_sleep(K_MSEC(10));
+		}
+
+		TERM_SUCCESS("Updating LL data length succeeded addr %s conn %u", addr,
+			     conn_ref_index);
+	}
+}
+#endif /* CONFIG_BT_USER_DATA_LEN_UPDATE */
+
+static void subscribe_to_service(struct bt_conn *conn, void *data)
+{
+	struct conn_info *conn_info_ref;
+
+	conn_info_ref = get_conn_info_ref(conn);
+	CHECKIF(conn_info_ref == NULL) {
+		TERM_WARN("Invalid reference returned");
+		return;
+	}
+
+	if (atomic_test_bit(conn_info_ref->flags, CONN_INFO_SUBSCRIBED_TO_SERVICE) == false) {
+		int err;
+		char addr[BT_ADDR_LE_STR_LEN];
+
+		bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+		memcpy(&uuid, PERIPHERAL_SERVICE_UUID, sizeof(uuid));
+		discover_params.uuid = &uuid.uuid;
+		discover_params.func = discover_func;
+		discover_params.start_handle = BT_ATT_FIRST_ATTRIBUTE_HANDLE;
+		discover_params.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+		discover_params.type = BT_GATT_DISCOVER_PRIMARY;
+
+		err = bt_gatt_discover(conn, &discover_params);
+		if (err) {
+			TERM_ERR("Discover failed(err %d)", err);
+			return;
+		}
+
+		while (atomic_test_bit(conn_info_ref->flags, CONN_INFO_SUBSCRIBED_TO_SERVICE) ==
+		       false) {
+			k_sleep(K_MSEC(10));
+		}
+	}
+}
+
+static void notify_peers(struct bt_conn *conn, void *data)
+{
+	int err;
+	struct bt_gatt_attr *vnd_ind_attr = (struct bt_gatt_attr *)data;
+	struct conn_info *conn_info_ref;
+
+	conn_info_ref = get_conn_info_ref(conn);
+	CHECKIF(conn_info_ref == NULL) {
+		TERM_WARN("Invalid reference returned");
+		return;
+	}
+
+	memset(vnd_value, 0x00, sizeof(vnd_value));
+	snprintk(vnd_value, notification_size, "%s%u", NOTIFICATION_DATA_PREFIX,
+		 conn_info_ref->tx_notify_counter);
+	err = bt_gatt_notify(conn, vnd_ind_attr, vnd_value, notification_size);
+	if (err) {
+		TERM_ERR("Couldn't send GATT notification");
+		return;
+	}
+
+	conn_info_ref->tx_notify_counter++;
+}
+
+void test_central_main(void)
+{
+	int err;
+	struct bt_gatt_attr *vnd_ind_attr;
+	char str[BT_UUID_STR_LEN];
+
+	memset(&conn_infos, 0x00, sizeof(conn_infos));
+
+	err = bt_enable(NULL);
+
+	if (err) {
+		TERM_ERR("Bluetooth init failed (err %d)", err);
+		return;
+	}
+
+	TERM_PRINT("Bluetooth initialized");
+
+	bt_gatt_cb_register(&gatt_callbacks);
+
+	vnd_ind_attr = bt_gatt_find_by_uuid(vnd_svc.attrs, vnd_svc.attr_count, &vnd_enc_uuid.uuid);
+
+	bt_uuid_to_str(&vnd_enc_uuid.uuid, str, sizeof(str));
+	TERM_PRINT("Indicate VND attr %p (UUID %s)", vnd_ind_attr, str);
+
+	start_scan();
+
+	while (true) {
+		while (atomic_test_bit(status_flags, BT_IS_SCANNING) == true ||
+		       atomic_test_bit(status_flags, BT_IS_CONNECTING) == true) {
+			k_sleep(K_MSEC(10));
+		}
+
+		if (conn_count < CONFIG_BT_MAX_CONN) {
+			start_scan();
+			continue;
+		}
+
+		if (check_all_flags_set(CONN_INFO_CONN_PARAMS_UPDATED) == false) {
```

Comment:
One of the test procedure requirements was to set the connection below 50 ms, so that was part of the test setup.
But, as you mentioned, we can continue while executing the connection parameters update procedure is running in the background.